sqlmesh/core/model/definition.py

```diff
@@ -2749,23 +2750,37 @@ def render_field_value(value: t.Any) -> t.Any:
 
         return value
 
+    def parse_strings_with_macro_refs(value: t.Any) -> t.Any:
+        if isinstance(value, str) and "@" in value:
+            return exp.maybe_parse(value, dialect=dialect)
+
+        if isinstance(value, dict):
+            for k, v in dict(value).items():
+                value[k] = parse_strings_with_macro_refs(v)
+        elif isinstance(value, list):
+            value = [parse_strings_with_macro_refs(v) for v in value]
+
+        return value
+
     for field_name, field_info in ModelMeta.all_field_infos().items():
         field = field_info.alias or field_name
+        field_value = fields.get(field)
 
-        if field in RUNTIME_RENDERED_MODEL_FIELDS:
+        if field in ("cron", "description") or field_value is None:
```

Comment:
We won't find this next time we introduce a similar field. Should we have a new named constant like `UNRENDERABLE_MODEL_FIELDS`

Reply:
Amended test and got rid of `description`, so just comparing to `cron` here now. Left a comment to explain why.